test/backend/shape_of.in.cpp

```diff
@@ -88,13 +140,39 @@ NGRAPH_TEST(${BACKEND_NAME}, shape_of_matrix)
     EXPECT_EQ(expected, read_vector<int64_t>(result));
 }
 
-NGRAPH_TEST(${BACKEND_NAME}, shape_of_5d)
+NGRAPH_TEST(${BACKEND_NAME}, shape_of_matrix_v3)
+{
+    Shape input_shape{2, 4};
+    Shape output_shape{2};
+
+    auto A = std::make_shared<op::Parameter>(element::f32, input_shape);
+    auto f =
+        std::make_shared<Function>(OutputVector{std::make_shared<op::v3::ShapeOf>(A),
+                                                std::make_shared<op::v3::ShapeOf>(A, element::i32)},
+                                   ParameterVector{A});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+
+    auto a = backend->create_tensor(element::f32, input_shape);
+    copy_data(a, vector<float>(2 * 4, 0));
+    auto result64 = backend->create_tensor(element::i64, output_shape);
+    auto result32 = backend->create_tensor(element::i32, output_shape);
+
+    auto handle = backend->compile(f);
+    handle->call_with_validate({result64, result32}, {a});
+    vector<int64_t> expected64{2, 4};
+    vector<int32_t> expected32{2, 4};
+    EXPECT_EQ(expected64, read_vector<int64_t>(result64));
+    EXPECT_EQ(expected32, read_vector<int32_t>(result32));
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, shape_of_5d_v0)
```

Comment:
@diyessi, I hope you will remove all such stuff before final merge. Is it for experiment only? opset3::ShapeOf should be committed and reviewed in a separate PR. Here we are massaging evaluate method and its automatic calling for all new ops in constant_folding pass and interpreter, right? And if we have all ingredients (except solution for separate/conditional compilation of reference kernel) I would like to finalize it and commit.

Reply:
More of the already merged ShapeOf3 PR.